include/jrl/Types.h

```diff
@@ -1,9 +1,11 @@
 #pragma once
+#include <gtsam/nonlinear/NonlinearFactorGraph.h>
 #include <gtsam/nonlinear/Values.h>
 
 namespace jrl {
 
 typedef std::map<gtsam::Key, std::string> ValueTypes;
+typedef std::function<bool(const std::string&, const gtsam::NonlinearFactor::shared_ptr&)> EntryPredicate;
```

Comment:
Since this is only used in the context of an entry lets move it within that class can rename it to FilterPredicate.
Externally it can be accessed with `Entry::FilterPredicate`.